### PR TITLE
approval pagination fix

### PR DESCRIPTION
**Pull request by garima468**
Approval pagination fix

### CODE REVIEW

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 1]
```diff
@@ -161,6 +163,8 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
 
       this.apprService.getApprovals(req).subscribe(res => {
         // this.data = []
+        this.totalprofileVerificationRecords = res.result.count
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: please put a check for res, if res and res.result like that

Comment (garima468) [reply project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: putted the condition

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 2]
```diff
@@ -161,74 +163,76 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
...
+          this.totaltransfersRecords = res.result.count
+          const newarray: any = []
+          let currentdate: Date
+          const resData = res.result.data
+          resData.forEach((approval: any) => {
+            // let keys = ''
+            approval.wfInfo.forEach((wf: any) => {
+              currentdate = new Date(wf.createdOn)
+              if (typeof wf.updateFieldValues === 'string') {
+                const fields = JSON.parse(wf.updateFieldValues)
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: Please do type check before parse

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 3]
```diff
@@ -161,74 +163,76 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
...
+                if (fields.length > 0) {
+                  fields.forEach((field: any) => {
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: Please do fields check

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 4]
```diff
@@ -161,74 +163,76 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
...
+                    // if (Object.keys(field.fromValue).length > 0) {
+                    //   keys += `${_.first(Object.keys(field.fromValue))}, `
+                    // } else {
+                    //   keys += `${_.first(Object.keys(field.toValue))}, `
+                    // }
+                    const labelKey = Object.keys(field.toValue)[0]
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: Please do check for field

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 5]
```diff
@@ -161,74 +163,76 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
...
-        this.allTransfersData = this.transfersData
-        this.allprofileVerificationData = this.profileVerificationData
-
-        if (this.profileVerificationData && this.profileVerificationData.length > 0) {
-          this.showApproveALL = true
-          this.disableApproveALL = false
-          // this.transfersCount = this.transfersData.length
+          newarray.forEach((appr: any) => {
+            const requestData = {
+              fullname: appr.userInfo ? `${appr.userInfo.first_name}` : '--',
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: appr.userInfo && app.userInfo.first_name

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 6]
```diff
@@ -161,74 +163,76 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
...
+              requestedon: currentdate,
+              // fields: this.replaceWords(keys, conditions),
+              userWorkflow: appr,
+              tag: (appr.userInfo && appr.userInfo.tag) ? `${appr.userInfo.tag}` : '',
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts:207]: Same as above

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts [hunk 7]
```diff
@@ -161,74 +163,76 @@ export class ApprovalPendingComponent implements OnInit, OnDestroy {
...
+            }
+            /* tslint:disable */
+            if (appr!.wfInfo[0] && appr!.wfInfo[0].orgTansferRequest) {
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: Please check for wfInfo

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts (hunk 2 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: fixed

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts (hunk 3 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: fixed

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts (hunk 4 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: fixed

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts (hunk 5 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: fixed

File: project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts (hunk 7 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/home/routes/approvals/approval-pending/approval-pending.component.ts]: fixed

**Pull request merged**